### PR TITLE
feat: use `preset-env` for appropriate client/server/modern targets

### LINKED ISSUE #110: Cannot access 'inject' before initialization

### PR DESCRIPTION
**Pull request by danielroe**
closes nuxt/nuxt.js#11682
resolves nuxt/bridge#110

**Future improvements**
- hook or some other way to configure targets

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,5 +28,11 @@
     "nuxt": "^2.15.3",
     "siroc": "^0.10.1",
     "standard-version": "^9.1.1"
+  },
+  "peerDependencies": {
+    "browserslist": "^4.0.0"
```
Comment (pi0) [inline package.json]: Using peerDependency means end-user or parent package should install this dependency but seems consumer is not our module? In this case maybe directly adding browserlist to dependencies?

File: src/index.ts [hunk 2]
```diff
@@ -1,34 +1,59 @@
...
-  } as Options)
-
-  nuxt.options.extensions.push('ts')
-  nuxt.options.build.additionalExtensions = ['ts', 'tsx']
+  } else {
+    coreJs = String(this.options.build.corejs)
+  }
 
-  nuxt.hook('webpack:config', (configs) => {
+  this.nuxt.hook('webpack:config', (configs) => {
```
Comment (pi0) [inline src/index.ts]: Can we please keep using `nuxt` variable extracted from this?

Comment (danielroe) [reply src/index.ts]: sure. just needed to move to `this.options` for type hinting, and thought it was simpler.

Comment (danielroe) [reply src/index.ts]: (I realise we will ultimately likely move to nuxt/nuxt.options when building this module with kit but it's a straightforward change to make at that point.)

Comment (danielroe) [reply src/index.ts]: cf: https://github.com/nuxt/nuxt.js/pull/9773

**Pull request merged**